source/Heuristics.cpp

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <mariana-trench/Heuristics.h>
+#include <mariana-trench/JsonValidation.h>
+
+namespace marianatrench {
+
+// Names of the heuristics parameters that can be set
```

Comment:
nit: we typically avoid defining const strings for json keys, just because it adds one more level of indirection when reading the code. Even more so if they're used within the same file (e.g. Model.cpp or TaintConfig.cpp).

Specifically, we'd just to the following at the call-site:

```
  if (JsonValidation::has_field(value, "k_join_override_threshold")) {
    heuristics.set_k_join_override_threshold(JsonValidation::unsigned_integer(
        value, "k_join_override_threshold"));
  }
```

Reply:
Solved in 54d1d44e9e1b4fbebcad31b6c1436c83cd968d71